### DISCUSSION TITLE
Agent visible in agents management but there's no asset inventory

### DISCUSSION THREAD

**Discussion by Sophal-Cam** [glpi-project/glpi-agent#306] [Q&A]
Hi,

Good day!

The agents are visible in agents' management tab but there's no computer asset linked/inventory.

Here is some info: 

GLPI 10.0.5
Plugin GLPI Inventory 1.0.6 
GLPI Agent 1.4.
_Here is the log issue I found in GLPI Agent Logfile_
[Thu Dec 29 08:23:31 2022][info] target server0: server http://XX.XX.XX.XX/glpi/front/inventory.php
[Thu Dec 29 08:23:31 2022][info] sending contact request to server0
[Thu Dec 29 08:23:31 2022][info] sending prolog request to server0
[Thu Dec 29 08:23:31 2022][info] running task Deploy
[Thu Dec 29 08:23:31 2022][error] [http client] communication error: 400 Bad Request
[Thu Dec 29 08:23:31 2022][info] Deploy task not supported by server0
[Thu Dec 29 08:23:34 2022][info] running task Inventory
[Thu Dec 29 08:23:34 2022][info] New inventory from pc.domain.local-2022-12-28-17-09-06 for server0
[Thu Dec 29 08:24:33 2022][error] [http client] communication error: 500 Internal Server Error
[Thu Dec 29 08:24:35 2022][info] running task Collect
[Thu Dec 29 08:24:35 2022][error] [http client] communication error: 400 Bad Request
[Thu Dec 29 08:24:35 2022][info] Collect task not supported by server0
[Thu Dec 29 08:24:37 2022][info] target server0: next run: Fri Dec 30 07:40:46 2022 - http://XX.XX.XX.XX/glpi/front/inventory.php
[Thu Dec 29 08:24:37 2022][info] GLPI Agent memory usage: WSS=3211264 PFU=154832896

What's wrong with it? How will I be able to see the asset!

Thanks in advance! :)
Sophal

**Comment by g-bougard** [comment]
Hi @Sophal-Cam 
you should probably enable debug=2 in registry and restart the service to check if something is wrong.
The `[error] [http client] communication error: 500 Internal Server Error` may mean something is not actually supported by native inventory.
You can eventually test with current nightly build: https://nightly.glpi-project.org/glpi-agent/
The nightly build has few update which can help understand what's going on with this http error 500.

**Reply by Sophal-Cam** [reply]
Hello @g-bougard 

Thanks for your advice and recommend. I have tested with nightly build with enabled debug=2 and I found the log error:
[Fri Dec 30 11:16:09 2022][info] GLPI Agent service starting
[Fri Dec 30 11:16:09 2022][debug] Trusted target ip: XX.XX.XX.XX/32, ::ffff:ac10:3d9/128
[Fri Dec 30 11:16:09 2022][debug] Trusted client ip/range: 127.0.0.1/32, ::ffff:7f00:1/128, ::1/128
[Fri Dec 30 11:16:09 2022][debug] [http server] Trying to load BasicAuthentication Server plugin
[Fri Dec 30 11:16:09 2022][debug] [basic authentication server plugin] Initializing BasicAuthentication v1.0 Server plugin...
[Fri Dec 30 11:16:09 2022][debug] [basic authentication server plugin] Loading BasicAuthentication Server plugin configuration from C:/Program Files/GLPI-Agent/etc/basic-authentication-server-plugin.cfg
[Fri Dec 30 11:16:09 2022][debug] [http server] HTTPD BasicAuthentication Server plugin loaded but disabled
[Fri Dec 30 11:16:09 2022][debug] [http server] Trying to load Inventory Server plugin
[Fri Dec 30 11:16:09 2022][debug] [inventory server plugin] Initializing Inventory v1.1 Server plugin...
[Fri Dec 30 11:16:09 2022][debug] [inventory server plugin] Loading Inventory Server plugin configuration from C:/Program Files/GLPI-Agent/etc/inventory-server-plugin.cfg
[Fri Dec 30 11:16:09 2022][debug] [http server] HTTPD Inventory Server plugin loaded but disabled
[Fri Dec 30 11:16:09 2022][debug] [http server] Trying to load Proxy Server plugin
[Fri Dec 30 11:16:09 2022][debug] [proxy server plugin] Initializing Proxy v2.1 Server plugin...
[Fri Dec 30 11:16:09 2022][debug] [proxy server plugin] Loading Proxy Server plugin configuration from C:/Program Files/GLPI-Agent/etc/proxy-server-plugin.cfg
[Fri Dec 30 11:16:09 2022][debug] [http server] HTTPD Proxy Server plugin loaded but disabled
[Fri Dec 30 11:16:09 2022][debug] [http server] Trying to load SecondaryProxy Server plugin
[Fri Dec 30 11:16:09 2022][debug] [proxy2 server plugin] Initializing SecondaryProxy v1.1 Server plugin...
[Fri Dec 30 11:16:09 2022][debug] [proxy2 server plugin] Loading SecondaryProxy Server plugin configuration from C:/Program Files/GLPI-Agent/etc/proxy2-server-plugin.cfg
[Fri Dec 30 11:16:09 2022][debug] [http server] HTTPD SecondaryProxy Server plugin loaded but disabled
[Fri Dec 30 11:16:09 2022][debug] [http server] Trying to load SSL Server plugin
[Fri Dec 30 11:16:09 2022][debug] [ssl server plugin] Initializing SSL v1.1 Server plugin...
[Fri Dec 30 11:16:09 2022][debug] [ssl server plugin] Loading SSL Server plugin configuration from C:/Program Files/GLPI-Agent/etc/ssl-server-plugin.cfg
[Fri Dec 30 11:16:09 2022][debug] [http server] HTTPD SSL Server plugin loaded but disabled
[Fri Dec 30 11:16:09 2022][debug] [http server] Trying to load Test Server plugin
[Fri Dec 30 11:16:09 2022][debug] [server test plugin] Initializing Test v1.5-gitc54d26f8 Server plugin...
[Fri Dec 30 11:16:09 2022][debug] [server test plugin] Loading Test Server plugin configuration from C:/Program Files/GLPI-Agent/etc/server-test-plugin.cfg
[Fri Dec 30 11:16:09 2022][debug] [http server] HTTPD Test Server plugin loaded but disabled
[Fri Dec 30 11:16:09 2022][debug] [http server] Trying to load ToolBox Server plugin
[Fri Dec 30 11:16:09 2022][debug] [toolbox plugin] Initializing ToolBox v1.1 Server plugin...
[Fri Dec 30 11:16:09 2022][debug] [toolbox plugin] Loading ToolBox Server plugin configuration from C:/Program Files/GLPI-Agent/etc/toolbox-plugin.cfg
[Fri Dec 30 11:16:09 2022][debug] [http server] HTTPD ToolBox Server plugin loaded but disabled
[Fri Dec 30 11:16:09 2022][info] [http server] HTTPD service started on port 62354
[Fri Dec 30 11:16:09 2022][debug2] Looking for Win32_Processor class WMI objects

I am not sure how to fix it. 

Thanks
Sophal

**Reply by g-bougard** [reply]
Hi @Sophal-Cam 
here this is just the normal log when the agent start. You have to force an inventory run or wait for to reach the point the agent contacts the server.

**Reply by Sophal-Cam** [reply]
Hi @g-bougard 

[glpi-agent.log](https://github.com/glpi-project/glpi-agent/files/10323823/glpi-agent.log)

Here is logfile after force an inventory run.

Thanks 
Sophal

**Reply by g-bougard** [reply]
I only see the same error. I think you have removed some lines, maybe too much. I don't see the server answer content.
Then you can try to generate an inventory and inject it this way:
```
glpi-inventory --json >inventory.json
glpi-injector -f inventory.json --verbose --debug -u http://xx.xx.xx.xx/glpi/front/inventory.php
```
You should run these commands from an administrative console and from the GLPI-Agent installation folder. Of course replace the url with your server url.
Then you should share all the output.

Alternatively, you also can monitor the `files/_log/php-errors.log` file on glpi server side to see if an error is logged by GLPI.

**Reply by Sophal-Cam** [reply]
Hello @g-bougard 

Here is info I got.
![image](https://user-images.githubusercontent.com/37643053/210072441-ff4d31de-6efe-4344-96c9-c407b3f85806.png)

And here is logfile I got
[glpi-agent.log](https://github.com/glpi-project/glpi-agent/files/10324811/glpi-agent.log)

Thanks
Sophal

**Reply by g-bougard** [reply]
Here "Unexpected empty server response" is clearly telling something goes wrong on server-side.
I can tell you this is not due to the inventory as I was able to import it in a test glpi.
Just a point, as you're using glpiinventory plugin you probably missed to follow that doc regarding your glpi-agent server url configuration: https://glpi-agent.readthedocs.io/en/latest/configuration.html#server

Anyway, you must investigate your GLPI server to find why it doesn't answer to glpi-agent. As I tell you before, take a look at the glpi server php-errors.log file.

P.S.: In your screenshot, I can see you trigerred an error related to batteries inventory. Feel free to open a specific issue for that.
